pkg/virt-api/rest/subresource.go

```diff
@@ -364,40 +367,36 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 		return
 	}
 
+	vmi, err := app.VirtCli.VirtualMachineInstance(namespace).Get(name, &k8smetav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			response.WriteError(http.StatusInternalServerError, err)
+			return
+		}
+		response.WriteError(http.StatusBadRequest, fmt.Errorf("VM is not running"))
+		return
+	}
```

Comment:
do we also want to check the status here, if the VM does not only exist, but is actually running?